### PR TITLE
Expose no-sign-up attribute on all elements (not only the login-fire-form)

### PR DESCRIPTION
**Pull request by JaySunSyn**
make noSignUp from login-form available in login-fire

### CODE REVIEW

File: login-fire-form.html [hunk 1]
```diff
@@ -298,6 +298,15 @@
         type: String
       },
 
+      /**
+       * If sign up suggestion should be visible.
+       *
+       * @type {Boolean}
+       */
+      noSignUp: {
```
Comment (masonlouchart) [inline login-fire-form.html]: The property `noSignUp` is already defined inside the `login-fire-form-behavior`.

**Review by masonlouchart — CHANGES_REQUESTED** [review]

File: login-fire-form.html (hunk 1 above)
Comment (JaySunSyn) [reply login-fire-form.html]: Hmm, I cannot see it... Do you want me to put it into the behavior?

Comment (JaySunSyn) [reply login-fire-form.html]: I looked in the wrong behavior ;)

**Comment by masonlouchart** [comment]
I figured out the `noSignUp` API was not forwarded to the `login-fire` element because the `login-fire-social` doesn't have it. If you need it we can add it but I think it must be documented on the `login-fire` element that this feature is only applied on the form. Or we have to implement it for `login-fire-socilal` too.

What do you think❓

**Comment by JaySunSyn** [comment]
I could expose `noSignUp` and link `noSignUp` to  `showSignUp` of `login-fire-social`.

So if someone sets `noSignUp`, `showSignUp` gets set to false.

What do you think?

**Comment by JaySunSyn** [comment]
It doesn't feel 100% clean though ... 

Hmm, what do you think should a `noSignUp` implementation in `login-fire-social` effect? Should it force `showSignUp` to false?

**Comment by masonlouchart** [comment]
The `login-fire-button` (used inside `login-fire-social`) has only "sign-in" and "sign-up" states. If as you suggested, the `noSignUp` property is linked to the `showSignUp` one, we could have render issue in the case `noSignIn` is True and `showSignUp` is false.

https://github.com/convoo/login-fire/blob/4ce55fab9f912d545c88572f433a0b628c7be689/login-fire-button.html#L275-L280

Actually, it should never been possible to have, for any of the elements, `showSignUp` set to False and `noSignIn` set to True. It means you don't need authentication at all or at least you don't allow signing-in nor sign-up...

__I guess, if we ensure this situation/case can't happen, we can apply the solution you proposed.__

__TODO:__
- [ ] be sure to expose `no-sign-up` attribute in all elements (it forbids signing-up)
- [ ] be sure to expose `no-sign-in` attribute in all elements (it forbids signing-in)
- [ ] be sure to expose `show-sign-up` attribute in all elements (it forces the element to render its "Sign Up View". Must be ignored when `noSignUp` is True)
- [ ] be sure we can't forbid signing-in and signing-up at the same time for any element (in this case, sign-in is available and a warning is printed in the console)

Some points of the TODO list are already done I guess but "guessing" is not allowed. We have to know 🤓

**Comment by masonlouchart** [comment]
Due to the huge refactoring on the branch that contains the next release, I suggest to close this PR nobody seems working on.

The issue shouldn't be in the next major release. Be careful to the breaking changes, especially the "sign-up, sign-in, sign-out" view selection that drastically changed.

**Comment by JaySunSyn** [comment]
Sure :) Looking forward to the next release.

**Comment by masonlouchart** [comment]
I'm on it!

**Pull request closed without merging**